controllers/datadogagent/feature/kubernetesstatecore/feature.go

```diff
@@ -96,12 +96,14 @@ func (f *ksmFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) feature.RequiredCom
 		f.enable = true
 		output.ClusterAgent.IsRequired = &f.enable
 
-		if dda.Spec.ClusterAgent.Config != nil && apiutils.BoolValue(dda.Spec.ClusterAgent.Config.ClusterChecksEnabled) {
-			f.clusterChecksEnabled = true
+		// tried to be align with the function "needClusterChecksRunner" https://github.com/DataDog/datadog-operator/blob/c07038fbf0b8993798ec5693cf6f191d7c369feb/controllers/datadogagent/clusterchecksrunner.go#L83-L92
+		if (apiutils.BoolValue(dda.Spec.ClusterChecksRunner.Enabled) || apiutils.BoolValue(dda.Spec.ClusterAgent.Config.ClusterChecksEnabled)) && apiutils.BoolValue(dda.Spec.Features.KubeStateMetricsCore.ClusterCheck) {
+			output.ClusterCheckRunner.IsRequired = apiutils.NewBoolPointer(true)
+
 			f.rbacSuffix = common.CheckRunnersSuffix
+			f.clusterChecksEnabled = true
```

Comment:
we should allow this to be `true` even if `dda.Spec.ClusterChecksRunner.Enabled` is `false`, right?